src/catalog/src/local/memory.rs

```diff
@@ -273,12 +281,13 @@ impl MemoryCatalogManager {
         }
     }
 
-    pub fn register_catalog_sync(&self, name: String) -> Result<bool> {
+    pub fn register_catalog_sync(self: Arc<Self>, name: String) -> Result<bool> {
```

Comment:
Accepting `self: &Arc<Self>` as the argument can reduce tons of `self.catalogs **.clone()** .register_catalog_sync(...)`. Also for `create_catalog_entry`.

Reply:
makes sense, since this is not a trait method